test/testUcumLhcUtils.spec.js

```diff
@@ -229,61 +229,287 @@ it("should return a message for invalid unit strings", function() {
+   */
+  it("should return 39.09 for a request to convert 1 eq to g with valance 1 and mw 39.09", function() {
+    var resp = utils.convertUnitTo('eq', 1, 'g', {
+      valence: 1,
+      molecularWeight: 39.09
+    });
+    assert.equal(resp.status, 'succeeded', resp.status + resp.msg);
+    assert.equal(resp.toVal.toPrecision(4), 39.09, resp.toVal.toPrecision(4));
+  });
+
+  /**
+   * CaCl2 -> Ca2+ + 2 Cl-
+   * Ca2+ has a valence of 2 and a molecular weight of 40.08
+   * ∴ 1 mol of Ca = 2 eq of Ca
+   * Equivalent mass of Ca2+ is its atomic mass divided by valence 2.
+   * Mass of calcium is 40.08 g/mole, so its equivalent mass is 40.08/2 =  20.04 g/equiv.  
+   */
+  it("should return 20.04 for a request to convert 1 eq to g with valance 2 and mw 40.08", function() {
+    var resp = utils.convertUnitTo('eq', 1, 'g', {
+      valence: 2,
+      molecularWeight: 40.08
+    });
+    assert.equal(resp.status, 'succeeded', resp.status + resp.msg);
+    assert.equal(resp.toVal.toPrecision(4), 20.04, resp.toVal.toPrecision(4));
+  });
+
+  /**
+   * CaCl2 -> Ca2+ + 2 Cl-
+   * Ca2+ has a valence of 2 and a molecular weight of 40.08
+   * ∴ 1 mol of Ca = 2 eq of Ca
+   * Equivalent mass of Ca2+ is its atomic mass divided by valence 2.
+   * Mass of calcium is 40.08 g/mole, so its equivalent mass is 40.08/2 =  20.04 g/equiv.  
+   * But we are converting 2 eq to g, so the result should be 40.08 g
+   */
+  it("should return 40.08 for a request to convert 2 eq to g with valance 2 and mw 40.08", function() {
+    var resp = utils.convertUnitTo('eq', 2, 'g', {
+      valence: 2,
+      molecularWeight: 40.08
+    });
+    assert.equal(resp.status, 'succeeded', resp.status + resp.msg);
+    assert.equal(resp.toVal.toPrecision(4), 40.08, resp.toVal.toPrecision(4));
+  });
+
+  /** Mass to eq conversion tests
+   * 
+   * The equivalent weight (EW) of an element is calculated as its atomic mass divided by its valence.
+   * EW = molecular weight / valence
+   * 
+   * The number of equivalents is then calculated by dividing the given mass by the equivalent weight.
+   * eq = amtMassOfSubstance / EW
+   */
+
+  /**
+   * K+ -> K+ 
+   * K+ has a valence of 1 and a molecular weight of 39.09
+   * ∴ 1 mol of K = 1 eq of K
+   * Equivalent mass of K+ is its atomic mass divided by valence 1.
+   * Mass of potassium is 39.09 g/mole, so its equivalent mass is 39.09/1 =  39.09 g/equiv.
+   * But we are converting 39.09 g to eq, so the result should be 40.08g/40.08g = 1 eq
+   */
+  it("should return 1 for a request to convert 39.09 g to eq with valance 1 and mw 39.09", function() {
+    var resp = utils.convertUnitTo('g', 39.09, 'eq', {
+      valence: 1,
+      molecularWeight: 39.09
+    });
+    assert.equal(resp.status, 'succeeded', resp.status + resp.msg);
+    assert.equal(resp.toVal.toPrecision(1), 1, resp.toVal.toPrecision(1));
+  });
+
+  /**
+   * CaCl2 -> Ca2+ + 2 Cl-
```

Comment:
A Chemistry question:  Suppose we had some pair of elements, A and B,  with AB -> A++ + B--.  (Is that possible?)  Then mole of A++ would react with 1 mole B--, so we would have A++ with a valence of 2, but, just 1 eq?

Reply:
While this condition is true, I believe the definition of an equivalent is the a amount of mol needed to balance the charge of 1 mol of an oppositely charged **monovalent**, which assumes a magnitude of 1. I'll confirm.

Not an element, but to address the example above:

1 MgSO<sub>4</sub> <-> 1 Mg<sup>2+</sup> + 1 SO<sub>4</sub><sup>2-</sup>